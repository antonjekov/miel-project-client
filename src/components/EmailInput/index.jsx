import React from 'react'
import styles from './index.module.css';
import { Form } from 'react-bootstrap';

function EmailInput({values, errors, handleChange}) {
    return (
        <Form.Group >
            <Form.Label>Email address</Form.Label>
            <Form.Control className={styles['form-control']} value={values.username} onChange={handleChange} type="email" placeholder="Enter email" name='username' isInvalid={!!errors.username} isValid={values.username && !errors.username} />
            <Form.Control.Feedback type='invalid' >{errors.username}</Form.Control.Feedback>
            <Form.Control.Feedback type='valid'>Look's good</Form.Control.Feedback>
        </Form.Group>
    )
}

export default EmailInput