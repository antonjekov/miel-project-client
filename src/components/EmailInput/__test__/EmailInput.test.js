import React from 'react'
import renderer from "react-test-renderer"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import EmailInput from '..'

describe('EmailInput', ()=>{

    afterEach(cleanup)
    it('matches snapshot without error', () => {
        const values = {username: 'anton@yahoo.com'}
        const errors = {}
        const tree = renderer.create( <EmailInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
    it('matches snapshot with error', () => {
        const values = {username: 'anton@yahoo.com'}
        const errors = {username: 'Some error'}
        const tree = renderer.create( <EmailInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })

})
