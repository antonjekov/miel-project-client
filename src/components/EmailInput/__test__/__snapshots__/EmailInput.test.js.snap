// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput matches snapshot with error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Email address
  </label>
  <input
    className="form-control form-control is-invalid"
    name="username"
    onChange={[Function]}
    placeholder="Enter email"
    type="email"
    value="anton@yahoo.com"
  />
  <div
    className="invalid-feedback"
  >
    Some error
  </div>
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;

exports[`EmailInput matches snapshot without error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Email address
  </label>
  <input
    className="form-control form-control is-valid"
    name="username"
    onChange={[Function]}
    placeholder="Enter email"
    type="email"
    value="anton@yahoo.com"
  />
  <div
    className="invalid-feedback"
  />
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;
